front/components/app/RootLayout.tsx

```diff
@@ -74,27 +75,29 @@ export default function RootLayout({
 
   return (
     <SparkleContext.Provider value={{ components: { link: NextLinkWrapper } }}>
-      <SWRConfig
-        value={{
-          onError: async (error) => {
-            if (
-              isAPIErrorResponse(error) &&
-              error.error.type === "not_authenticated"
-            ) {
-              // Redirect to login page.
-              await router.push("/api/auth/login");
-            }
-          },
-        }}
-      >
-        <UserProvider>
-          <SidebarProvider>
-            <ConfirmPopupArea>
-              <Notification.Area>{children}</Notification.Area>
-            </ConfirmPopupArea>
-          </SidebarProvider>
-        </UserProvider>
-      </SWRConfig>
+      <ThemeProvider>
```

Comment:
As far as I know the order doesn't matter here? I put it somewhere close to SparkleContext since it's kind of design related??